### ISSUE TITLE
use gravity as setpoint in ferm profile

### ISSUE THREAD

**Issue by brendongough** [thorrak/fermentrack#287]
In additional to the current time based setpoint, allow for a gravity based setpoint.
There's a lot of factors which could change expected fermentation throughout a time frame however the gravity measurement is a true indication.
This would be particularly useful for stepped fermentation profiles; fast ferment lager or using it to do a diacetyl rest close to terminal gravity.
A check could be added prior to assigning a ferm profile to a controller to ensure it has a gravity measurement before allowing a user to apply a specific profile.

**Comment by thorrak** [comment]
This is coming. The goal is to enable three types of gravity-based profiles:

Gravity targets (Hold temp until 1.030)
Attenuation targets (Hold temp until 30% attenuated)
Phase targets (Hold temp until end of lag phase)

Gravity targets are the simplest to implement, phase targets may turn out to be more difficulty than its worth - but the goal is to have all 3. 

These will be implemented after #93 and will probably require some refactoring of the fermentation profile code.

**Comment by gizzygizmo** [comment]
one note... the profiles should be able to have an "AND/OR" ability.  There is a possibility of range issues, battery issues, network issues, etc.   If your profile only progresses to next step when "gravity" breaks a set-point it may be stuck forever if we've lost communications.  Or worse, if a sudden spike causes your fermenter to d-rest during turbulent fermentation.

"AND" would be good for "3days" and "Gravity below 1.020" to cause diacetyl rest
"OR" would be good for similar logic if you trust tilt better but want to ensure that at least day7 we have rest -> "7 days or below 1.020"

* AND as well as OR operator for conditions at set-point changes
* make set-point for gravity be met by 3x or 5x consecutive updates to avoid bad reading due to moving carboy or turbulent fermentation

**Comment by thorrak** [comment]
Agreed

**Comment by uSlackr** [comment]
Agreed.  Perhaps the gravity reading should be for some period of time – gravity below 1.010 for 4 hours or some such.

\\Greg<file://Greg>

**Comment by thorrak** [comment]
For what it's worth, I'm leaning towards targeting this next after I get Docker support released. I've got a lot of Tilt-related changes that I'm working on behind the scenes - this feels like a good time to get this in.

**Comment by gizzygizmo** [comment]
Another thought... while setting up a lager today.  There is no "free rise" capability.   Would be nice to disable the heater function in the beer profile so I could set a max temp and let the beer free rise.

For example, i would normally pitch at 50degF and let it free rise to 55degF.   If I try this with a profile i have to set a 1d or 2d at "55" and let it ramp... but fermentrack will actually use the heater to get it there... rather than a real free rise.

Maybe an indicator that you're setting "free rise" and that means the cooling will only be activated and the heater will be disabled until the next setpoint.
